components/net/http_loader.rs

```diff
@@ -682,7 +673,8 @@ pub fn http_redirect_fetch(request: &mut Request,
     request.redirect_count += 1;
 
     // Step 7
-    let same_origin = location_url.origin()== request.current_url().origin();
+    // FIXME: Correctly use request's origin
+    let same_origin = location_url.origin() == request.current_url().origin();
```

Comment:
Let's use the correct origin here. We can [create](https://doc.servo.org/servo_url/origin/enum.ImmutableOrigin.html#method.new) an ImmutableOrigin value from the url's origin and compare them.